lib/client/api.js

```diff
@@ -281,7 +283,16 @@ API.prototype.createAddress = function(cb) {
   var data = this._loadAndCheck();
 
   var url = '/v1/addresses/';
-  this._doPostRequest(url, {}, data, cb);
+  this._doPostRequest(url, {}, data, function(err, address) {
+    if (err) return cb(err);
+
+    if (data.publicKeyRing.length != data.n)
+      return cb('Wallet Incomplete, cannot derive address.')
+
+    Verifier.checkAddress(data, address);
```

Comment:
should be `if (Verifier.checkAddress(data, address)) return cb('Server sent fake address');`